### DISCUSSION TITLE
Share fragment between query and mutation without masking

### DISCUSSION THREAD

**Discussion by KammererTob** [dotansimha/graphql-code-generator#9325] [Q&A]
To avoid duplication i would like to use the same fragment for a query and a mutation.
Example:

```
graphql(/* GraphQL */`
   fragment SettingsFragment on Settings {
     id
     name
     user
   }
`)

const SettingsQuery = graphql(/* GraphQL */ `
  query SettingsQuery($id: ID!) {
    settings: settings(id: $id) {
      ...SettingsFragment
    }
  }
`)

const SettingsMutation = graphql(/* GraphQL */`
  mutation SaveSettings($settings: SettingsInput!) {
    settings: saveSettings(settings: $settings) {
      ...SettingsFragment
    }
  }
`)

const { result } = useQuery(SettingsQuery, { id: props.id } )
```

The issue with this approach is that the fragment is masked in the component. Meaning that i cannot access `result.value?.settings.name`, i can only access `__typename`. I assume that is because i am using a fragment here. What is the best approach here?

**Comment by hmz22** [comment]
you just need disable fragment masking.
```
...
  generates: {
    "src/common/graphql/types/": {
      preset: "client",
+     presetConfig: {
+       fragmentMasking: false,
+     },
      plugins: [],
    },
  },
...
```

**Reply by KammererTob** [reply]
I actually want fragmentMasking, but not in this instance since they both are defined in the same component.

**Reply by FlawaCLV** [reply]
Not working for me, I still have $fragmentRefs

**Comment by SPAHI4** [comment]
Any solutions found?
